File: credit.c
#include <stdio.h>

int main(void)
{
    int sum1 = 0, sum2 = 0, total = 0;
    int rem = 0, div = 0;
    int digit = 0;
    int last, second_last;
    long long int card, a, b, n, start;
    printf("Enter your Card Number: ");
    scanf("%lld",&card);
    start = card;

    // Check Card length
    n = card;
    do
    {
        n = n / 10;
        digit++;
    }
    while (n > 0);


    if (digit != 13 && digit != 15 && digit != 16)
    {
        printf("Card is INVALID\n");
        return 0;
    }


    //Check whether the card is valid
    while (card != 0)
    {
        //for the last digit
        last = card % 10;
        card = card / 10;
        sum1 = sum1 + last;

        //for the second last digit
        second_last = card % 10;
        card = card / 10;

        //Double second last digit
        second_last = second_last * 2;
        rem = second_last % 10;
        div = second_last / 10;
        sum2 = sum2 + rem + div;
    }
    total = sum1 + sum2;


    //Check Luhn's Algorithim
    if (total % 10 != 0)
    {
        printf("Card is INVALID\n");
        return 0;
    }


    //Getting the start numbers
    do
    {
        start = start / 10;
    }
    while (start > 100);


    //For MASTER CARD
    if (50 < start && start < 56)
    {
        printf("MASTERCARD\n");
    }

    //For AMERICAN EXPRESS
    else if (start == 34 || start == 37)
    {
        printf("AMEX\n");
    }

    //For VISA
    else if (start / 10 == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py
# initializing or declaring variables
u = 0
sum1 = 0
sum2 = 0
    
# making sure the number is positive
while True:
    try:
        card = int(input("Number: "))
        if card > 0:
            break
    except ValueError:
        card = int(input("Number: "))

card1 = card
card3 = card
card2 = card1 // 10
    
# multiplying every other digit by 2, starting from second last
while card2 > 0:
    p = card2 % 10
    p = p * 2
    card2 = card2 // 100
    if p > 9:
        q = p % 10
        p = p // 10
        r = p % 10
        p = q + r
    sum1 = sum1 + p

# adding every other digit starting from last
while card1 > 0:
    s = card1 % 10
    card1 = card1 // 100
    sum2 = sum2 + s

sum = sum1 + sum2

# to obtain the last digit
t = sum % 10

# the number of digits
while card3 > 0:
    card3 = card3 // 10
    u += 1

# obtaining the first and first two digits
x = card // pow(10, (u - 1))
y = card // pow(10, (u - 2))

# checking whether the card number is valid or not
if t == 0:
    # for AMEX
    if (u == 15 and (y == 34 or y == 37)):
        print("AMEX")

    # for MASTERCARD
    elif (u == 16 and (y > 50 and y < 56)):
        print("MASTERCARD")

    # for VISA
    elif ((u == 16 or u == 13) and x == 4):
        print("VISA")
    
    # if the card is not valid
    else:
        print("INVALID")
        
else:
    print("INVALID")